backend/dal/spc_charts_dal.py

```
@@ -485,6 +456,128 @@ async def fetch_chart_data_values(
+    query = f"""
+        SELECT
+            MEASURE(mean_value)    AS cl,
+            MEASURE(x_bar_ucl)     AS ucl,
+            MEASURE(x_bar_lcl)     AS lcl,
+            MEASURE(sigma_within)  AS sigma_within,
+            MEASURE(cpk)           AS cpk,
+            MEASURE(ppk)           AS ppk
+        FROM {tbl('spc_quality_metrics')}
+        {where_sql}
+        GROUP BY mic_id
+    """
```

Ruff (0.15.10)

[error] Possible SQL injection vector through string-based query construction

(S608)